### PR TITLE
feat(java): add adoptium eclipse temurin 17

### PR DESCRIPTION
**Pull request by kristofferwanglund**

### CODE REVIEW

**Comment by Edholm** [comment]
Is this something you feel will be useful to a "larger" audience? If only specific for one or two repos I would suggest adding it locally to those instead.

File: tools/sggit/tools.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ func VerifyNoDiff(ctx context.Context) error {
 	if status.String() != "" {
 		dirtyDetails := status.String()
 		// attempt to give a nice patch output if the dirty files are tracked by git
-		if patchOutput := sg.Output(Command(ctx, "diff", "--patch")); patchOutput != "" {
+		if patchOutput := sg.Output(Command(ctx, "diff", "--name-only")); patchOutput != "" {
```
Comment (Edholm) [inline tools/sggit/tools.go:29]: This might break things in unexpected ways so let's not include in this PR :)

Comment (kristofferwanglund) [reply tools/sggit/tools.go:29]: Good idea, it was just something I have thought of.

**Comment by kristofferwanglund** [comment]
@Edholm good question. It is for one specific repo so I could add it there. However, Sage already has maven support and maven requires Java so I think this change makes Sage more complete.

File: tools/sggit/tools.go (hunk 1 above)
Comment (kristofferwanglund) [reply tools/sggit/tools.go]: Fixed now.

**Pull request closed without merging**